app/cmd/routes.go

```diff
@@ -159,6 +159,7 @@ func routes(r *web.Engine) *web.Engine {
 		api.Use(middlewares.IsAuthenticated())
 
 		api.Post("/api/v1/posts", apiv1.CreatePost())
+                api.Get("/api/v1/posts/:number", apiv1.GetPost())
```

Comment:
Wow, how did I miss this API? Thank you!

Few comments:

1. We can move it before the IsAuthenticated middleware, right next to  `/api/v1/posts`
2. Align it with the rest of the code.